extensions/applicationinsights-dependencies-js/src/ajax.ts

```diff
@@ -50,749 +141,685 @@ export class AjaxMonitor extends BaseTelemetryPlugin implements IDependenciesPlu
             enableCorsCorrelation: false,
             enableRequestHeaderTracking: false,
             enableResponseHeaderTracking: false,
-            enableAjaxErrorStatusText: false
+            enableAjaxErrorStatusText: false,
+            enableAjaxPerfTracking: false,
+            maxAjaxPerfLookupAttempts: 3,
+            ajaxPerfLookupDelay: 25
         }
         return config;
     }
 
     public static getEmptyConfig(): ICorrelationConfig {
-        return {
-            maxAjaxCallsPerView: undefined,
-            disableAjaxTracking: undefined,
-            disableFetchTracking: undefined,
-            disableCorrelationHeaders: undefined,
-            distributedTracingMode: undefined,
-            correlationHeaderExcludedDomains: undefined,
-            appId: undefined,
-            enableCorsCorrelation: undefined,
-            correlationHeaderDomains: undefined,
-            enableRequestHeaderTracking: undefined,
-            enableResponseHeaderTracking: undefined,
-            enableAjaxErrorStatusText: undefined
-        }
-    }
-
-    private static getFailedAjaxDiagnosticsMessage(xhr: XMLHttpRequestInstrumented): string {
-        let result = "";
-        try {
-            if (!_isNullOrUndefined(xhr) &&
-                !_isNullOrUndefined(xhr.ajaxData) &&
-                !_isNullOrUndefined(xhr.ajaxData.requestUrl)) {
-                result += "(url: '" + xhr.ajaxData.requestUrl + "')";
-            }
-        } catch (e) { }
+        let emptyConfig = this.getDefaultConfig();
+        _arrForEach(CoreUtils.objKeys(emptyConfig), (value) => {
+            emptyConfig[value] = undefined;
+        });
 
-        return result;
+        return emptyConfig;
     }
+
     public identifier: string = AjaxMonitor.identifier;
 
     priority: number = 120;
-    protected initialized: boolean; // ajax monitoring initialized
-    protected _fetchInitialized: boolean; // fetch monitoring initialized
-    protected _config: ICorrelationConfig = AjaxMonitor.getEmptyConfig();
-    protected _trackAjaxAttempts: number = 0;
-    private currentWindowHost: string;
-    private _context?: ITelemetryContext;
-    private _isUsingW3CHeaders: boolean;
-    private _isUsingAIHeaders: boolean;
 
     constructor() {
         super();
         let location = getLocation();
-        this.currentWindowHost = location && location.host && location.host.toLowerCase();
-        this.initialized = false;
-        this._fetchInitialized = false;
-    }
-
-    trackDependencyData(dependency: IDependencyTelemetry, properties?: { [key: string]: any }) {
-        this.trackDependencyDataInternal(dependency, properties);
-    }
-
-    public processTelemetry(item: ITelemetryItem, itemCtx?: IProcessTelemetryContext) {
-        this.processNext(item, itemCtx);
-    }
+        let _fetchInitialized:boolean = false;      // fetch monitoring initialized
```

Comment:
what's the difference to put these variables in constructor than leave outside? 🤔

Reply:
Outside the constructor they become object (this) level properties and therefore all references will need to address them as this.xxxxx

This leads to 2 problems (especially for private / internal properties).
1) The names can no longer be minified and must always be referenced as this.xxx (or self.xxxx)
2) They are now exposed and can be altered externally, part of the point of private properties is they should be internal to the implementation / class.

By moving them into the constructor the "scope" (visibility) of these variables is now limited to the within the constructor and any and all functions defined within the constructor.

Shorter answer
- Better minification
- Hiding private / internal properties, only used to track internal state.